pkg/validatorfactory/factory.go

```diff
@@ -196,15 +194,20 @@ func removeRefs(defs map[string]*spec.Schema, sch spec.Schema) spec.Schema {
 	// we squash it so that the Ref is direct without AllOf
 
 	if len(sch.AllOf) == 1 && len(sch.AllOf[0].Ref.String()) > 0 {
-		vCopy := sch
-		vCopy.AllOf = nil
-		vCopy.Default = nil
-		vCopy.Example = nil
-		vCopy.Description = ""
-
-		if reflect.DeepEqual(vCopy, spec.Schema{}) {
-			return removeRefs(defs, sch.AllOf[0])
+		vCopy := removeRefs(defs, sch.AllOf[0])
+		vCopy.Default = sch.Default
+		vCopy.Description = sch.Description
+		if sch.Extensions != nil {
```

Comment:
probably want to copy all nonzero properties and not just Default, Description, Extensions. Not sure if there is a clean go-ey way to do that